### PR TITLE
iTerm2: 3.4.0 -> 3.4.8

### PR DESCRIPTION
**Pull request by VergeDX**
###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/terminal-emulators/iterm2/default.nix [hunk 1]
```diff
@@ -42,7 +48,7 @@ stdenv.mkDerivation rec {
     homepage = "https://www.iterm2.com/";
     license = licenses.gpl2;
     maintainers = with maintainers; [ tricktron ];
-    platforms = platforms.darwin;
+    platforms = [ "x86_64-darwin" ];
```
Comment (hexagonal-sun) [inline pkgs/applications/terminal-emulators/iterm2/default.nix:51]: Does this change prevent `iterm2` running on Apple silicon?

Comment (VergeDX) [reply pkgs/applications/terminal-emulators/iterm2/default.nix:51]: Oh if I keep platforms.darwin, the GitHub Action for this PR will fail.
Seems it's dependence missing on aarch64-darwin, so I delete it.

Comment (SuperSandro2000) [reply pkgs/applications/terminal-emulators/iterm2/default.nix:51]: yes

**Review by fufexan — APPROVED** [review]

File: pkgs/applications/terminal-emulators/iterm2/default.nix (hunk 1 above)
Comment (hexagonal-sun) [reply pkgs/applications/terminal-emulators/iterm2/default.nix:51]: I reverted this line to:

```
platforms.darwin
```

and it built fine on my M1 machine. If we could leave this to include apple silicon that would be great!

**Pull request closed without merging**